source/app/iris_engine/module_handler/module_handler.py

```diff
@@ -186,39 +235,283 @@ def register_module(module_name):
+            manual_hook_name = f"{module.module_name}::{iris_hook_name}"
+
+    hook = IrisHook.query.filter(IrisHook.hook_name == iris_hook_name).first()
+    if not hook:
+        return False, [f"Hook {iris_hook_name} is unknown"]
+
+    if not isinstance(is_manual_hook, bool):
+        return False, [f"Expected bool for is_manual_hook but got {type(is_manual_hook)}"]
+
+    if not isinstance(run_asynchronously, bool):
+        return False, [f"Expected bool for run_asynchronously but got {type(run_asynchronously)}"]
+
+    mod = IrisModuleHook.query.filter(
+        IrisModuleHook.hook_id == hook.id,
+        IrisModuleHook.module_id == module_id,
+        IrisModuleHook.manual_hook_ui_name == manual_hook_name
+    ).first()
+    if not mod:
+        imh = IrisModuleHook()
+        imh.is_manual_hook = is_manual_hook
+        imh.wait_till_return = False
+        imh.run_asynchronously = run_asynchronously
+        imh.max_retry = 0
+        imh.manual_hook_ui_name = manual_hook_name
+        imh.hook_id = hook.id
+        imh.module_id = module_id
+
+        try:
+            db.session.add(imh)
+            db.session.commit()
+        except Exception as e:
+            return False, [str(e)]
+
+        return True, [f"Hook {iris_hook_name} registered"]
+
+    else:
+        return True, [f"Hook {iris_hook_name} already registered"]
+
+
+def deregister_from_hook(module_id: int, iris_hook_name: str):
+    """
+    Deregister from an existing hook. The hook_name should be a well-known hook to IRIS. No error are thrown if the
+    hook wasn't register in the first place
+
+    :param module_id: Module ID to deregister
+    :param iris_hook_name: hook_name to deregister from
+    :return: IrisInterfaceStatus object
+    """
+    log.info(f'Deregistering module #{module_id} from {iris_hook_name}')
+    hooks = IrisModuleHook.query.filter(
+        IrisModuleHook.module_id == module_id,
+        IrisHook.hook_name == iris_hook_name,
+        IrisModuleHook.hook_id == IrisHook.id
+    ).all()
+    if hooks:
+        for hook in hooks:
+            log.info(f'Deregistered module #{module_id} from {iris_hook_name}')
+            db.session.delete(hook)
+
+    return True, ['Hook deregistered']
+
+
+@celery.task(bind=True)
+def task_hook_wrapper(self, module_name, hook_name, hook_ui_name, data, init_user, caseid):
+    """
+    Wrap a hook call into a Celery task to run asynchronously
+
+    :param self: Task instance
+    :param module_name: Module name to instanciate and call
+    :param hook_name: Name of the hook which was triggered
+    :param hook_ui_name: Name of the UI hook so module knows which hook was called
+    :param data: Data associated to the hook to process
+    :param init_user: User initiating the task
+    :param caseid: Case associated
+    :return: A task status JSON task_success or task_failure
+    """
+    # Data is serialized, so deserialized
+    deser_data = loads(data=base64.b64decode(data))
```

Comment:
loads takes a positionnal argument, so it expects to be called as `loads(base64.b64decode(data))`. As it is, it raises a TypeError:
```
TypeError: loads() takes exactly 1 positional argument (0 given)
```